::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Fri, 8 Apr 2016 15:28:43 +0200
Subject: [PATCH] FIXME: Add --force-yes to apt-get dist-upgrade.

arm64 udev in Ubuntu Trusty is broken, so we had to provide our own
patched udev package in armband MOS repos.

Due to dpkg version comparison algorithm, our MOS version of udev
is considered a downgrade, which requires --force-yes for
apt-get dist-upgrade to work and pick up this version, otherwise
bootstrap/target image build would fail with apt-get error code 100.

This change can be dropped later, if other packages do not manifest
the same behavior.
---
 fuel_agent/utils/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fuel_agent/utils/build.py b/fuel_agent/utils/build.py
index 5761cc5..5557810 100644
--- a/fuel_agent/utils/build.py
+++ b/fuel_agent/utils/build.py
@@ -107,7 +107,7 @@ def run_apt_get(chroot, packages, eatmydata=False, attempts=10):
     time in X times.
     """
     for action in ('update', 'dist-upgrade'):
-        cmds = ['chroot', chroot, 'apt-get', '-y', action]
+        cmds = ['chroot', chroot, 'apt-get', '-y', '--force-yes', action]
         stdout, stderr = utils.execute(*cmds, attempts=attempts)
         LOG.debug('Running apt-get %s completed.\nstdout: %s\nstderr: %s',
                   action, stdout, stderr)
